src/compiler/crystal/codegen/primitives.cr

```diff
@@ -612,6 +612,14 @@ class Crystal::CodeGenVisitor
       end
       arg
     when from_type.rank < to_type.rank
+      # extending a signed integer to an unsigned one (eg: Int8 to UInt16)
+      # may still lead to underflow
+      if checked
+        if from_type.signed? && to_type.unsigned?
+          overflow = codegen_out_of_range(to_type, from_type, arg)
+          codegen_raise_overflow_cond(overflow)
```

Comment:
Somehow I feel the code below this shouldn't execute... 🤔

Reply:
I don't follow the comment @asterite. What do you mean?

Reply:
Two lines below this there's more code and it will be generated, but it shouldn't. This line is missing a return.

Reply:
oh, maybe not? I thought the check would always raise... nevermind

Reply:
The codegen will emit a raise block and leave the builder in place to continue the normal path which in this case is the extend_int. But this method itself won't raise. It's generated code will.